frontend/src/components/MapPanel.tsx

```diff
@@ -116,14 +120,25 @@ function MapPanel({
       setupMapInteractions(map);
       updateDataSources(reportResults, relationshipResults, box, isSearchMode, map);
     }
+
+    setTotalDataPoints(reportResults.length + relationshipResults.length);
   }, [reportResults, relationshipResults]);
 
   return (
     <div className="map-panel-container">
       <div className="map" ref={mapContainerRef} />
       {isSearchMode
         ? (
-          <button className="search-button" type="button" onClick={updateSearch}>Search Area</button>
+          <>
+            <button className="search-button" type="button" onClick={updateSearch}>Search Area</button>
+            { (reportResults.length > 0 || relationshipResults.length > 0)
```

Comment:
Small tip, you could use:
`{(reportResults.length > 0 || relationshipResults.length > 0) && <Component/>}` rather than ternary I think?